### PR TITLE
Add client-side rollback during snapshot import

### LINKED ISSUE #490: Add client-side rollback during snapshot import
**Proposed feature**
When importing a snapshot, there's currently no way to roll back on errors client-side. To make it more robust, a "clean" snapshot should be saved before deleting nodes and importing the new ones, allowing a rollback to that clean snapshot in case of any errors.

### PR DESCRIPTION
**Pull request by sorenjohanson**
closes #490

### CODE REVIEW

File: teammapper-frontend/mmp/src/map/handlers/history.ts [hunk 1]
```diff
@@ -120,8 +125,11 @@ export default class History {
      * Redraw the map with a new snapshot.
      * @param {MapSnapshot} snapshot
      */
-    private redraw(snapshot: MapSnapshot) {
-        this.map.nodes.clear()
+    private redraw(snapshot: MapSnapshot, backup: boolean = false) {
```
Comment (JannikStreek) [inline teammapper-frontend/mmp/src/map/handlers/history.ts:128]: so when is this called? currently its not? its always false, isn't it?

**Comment by JannikStreek** [comment]
How can I test this locally?

File: teammapper-frontend/mmp/src/map/handlers/history.ts (hunk 1 above)
Comment (sorenjohanson) [reply teammapper-frontend/mmp/src/map/handlers/history.ts:128]: I forgot to add `this.redraw(snapshot, true)` to the snapshot import. :D 

This is only called during a snapshot import right now.

**Comment by sorenjohanson** [comment]
Apparently still buggy, so I reset this to draft for now.

**Comment by sorenjohanson** [comment]
The rollback uses `redraw` directly now without relying on `save()`, this should allow for a more robust undo and prevent stale snapshots (which was an issue before).

**Pull request merged**